webui/react/src/components/LogViewer.tsx

```diff
@@ -327,9 +326,37 @@ const LogViewer: React.FC<Props> = forwardRef((
     }
   }, [ config, scrollToInfo ]);
 
+  useLayoutEffect(() => {
+    const handleCopy = (e: ClipboardEvent): void => {
+      const clipboardFormat = 'text/plain';
+      const selection = window.getSelection()?.toString() || '';
+      const lines = selection?.split('\n');
+      if (lines?.length <= 1) {
+        e.clipboardData?.setData(clipboardFormat, selection);
+      } else {
+        const oddOrEven = lines.map(line => /^\[/.test(line) || /\]$/.test(line))
+          .reduce((acc, isTimestamp, index) => {
+            if (isTimestamp) acc[index % 2 === 0 ? 'even' : 'odd']++;
+            return acc;
+          }, { even: 0, odd: 0 });
+        const isEven = oddOrEven.even > oddOrEven.odd;
+        const content = lines.reduce((acc, line, index) => {
+          const skipNewline = (isEven && index % 2 === 0) || (!isEven && index % 2 === 1);
+          return acc + line + (skipNewline ? ' ' : '\n');
+        }, '');
+        e.clipboardData?.setData(clipboardFormat, content);
```

Comment:
this is basically assuming timestamp -> newline -> log line format right?  are there never multiple log lines between timestamps?

this could use a comment explaining why we're doing this